wrench/src/perf.rs

```diff
@@ -188,30 +190,28 @@ impl<'a> PerfHarness<'a> {
         let draw_calls = cpu_frame_profiles[0].draw_calls;
         assert!(cpu_frame_profiles.iter().all(|s| s.draw_calls == draw_calls));
 
-        cpu_frame_profiles.sort_by_key(|a| a.composite_time_ns);
-        gpu_frame_profiles.sort_by_key(|a| a.paint_time_ns);
-
-        // Remove the two slowest and fastest frames.
-        let cpu_samples = &cpu_frame_profiles[2..cpu_frame_profiles.len() - 2];
-        let gpu_samples = &gpu_frame_profiles[2..gpu_frame_profiles.len() - 2];
-
-        // Use the mean value from the remaining frames.
-        let composite_time: u64 = cpu_samples.iter()
-                                             .map(|s| s.composite_time_ns)
-                                             .sum();
-        let paint_time: u64 = gpu_samples.iter()
-                                         .map(|s| s.paint_time_ns)
-                                         .sum();
+        let composite_time_ns = extract_sample(&mut cpu_frame_profiles, |a| a.composite_time_ns);
+        let paint_time_ns = extract_sample(&mut gpu_frame_profiles, |a| a.paint_time_ns);
+        let backend_time_ns = extract_sample(&mut cpu_frame_profiles, |a| a.backend_time_ns);
 
         TestProfile {
             name: filename.to_str().unwrap().to_string(),
-            composite_time_ns: composite_time / cpu_samples.len() as u64,
-            paint_time_ns: paint_time / gpu_samples.len() as u64,
+            composite_time_ns: composite_time_ns,
```

Comment:
nit: looks like you haven't gotten used to the new shiny Rust 1.16 features yet :)